tests/run-make/exit-code/rmake.rs

```diff
@@ -0,0 +1,53 @@
+extern crate run_make_support;
+
+use run_make_support::{rustc, rustdoc, tempdir};
+
+fn main() {
+    rustc()
+        .arg("success.rs")
+        .run();
+
+    assert_eq!(rustc()
+        .arg("--invalid-arg-foo")
+        .run_fail()
+        .status
+        .code().unwrap(), 1);
```

Comment:
If you use `run_fail` which calls `handle_failed_output`, I believe you don't even need to assert the exit code (because `handle_failed_output` will call `std::process::exit(1)`). So this can just be

```rs
rustc()
  .arg("--invalid-arg-foo")
  .run_fail()
```

same with the other `run_fail`. The `.status.code().unwrap()` and its outer assert won't be reachable.

Reply:
We *do* want to assert the exit code is 1, so if this extra code isn't reachable with the test as-is that's a bug in the test, not just dead code. (The test's purpose is to check we're specifically failing with 1, not 101 or some other exit code).

Reply:
Don't we want to assert the exit code is 1 (as opposed to some other number)? The ICE test checks specifically for an ICE exit code. I don't know if we consider the specific exit codes here stable/worth testing for. But FWIW the old tests do check specifically for 1.

(could make `run_fail` take an expected exit code instead, I guess? That'd clean up the code)

Reply:
> We _do_ want to assert the exit code is 1, so if this extra code isn't reachable with the test as-is that's a bug in the test, not just dead code. (The test's purpose is to check we're specifically failing with 1, not 101 or some other exit code).

You're right, the test does need to check for the specific exit code of `1`. In this case, I think we might be better off with a helper method of `run_fail_expect_exit_code()` or something so other tests that don't specifically need to match on a specific exit code doesn't need to pass it in?